[RHACS] ROX11620 Updating RHACS diagnostic command (#56709)
openshift/openshift-docs: modules/generate-diagnostic-bundle-using-roxctl-cli.adoc

:_content-type: PROCEDURE
[id="generate-diagnostic-bundle-using-roxctl-cli_{context}"]
= Generating a diagnostic bundle by using the roxctl CLI

You can generate a diagnostic bundle by using the `roxctl` CLI.

.Prerequisites
* To generate a diagnostic bundle, you need `read` permission for the `Administration` resource. This is required for versions of the `DebugLogs` resource older than version 3.73.0.

* You have configured the `ROX_PASSWORD` and `ROX_CENTRAL_ADDRESS` environment variables.

** Run the following command to set the `ROX_PASSWORD` and `ROX_CENTRAL_ADDRESS` environment variables:
+
[source,terminal,subs="+quotes"]
----
$ export ROX_PASSWORD=_<rox_password>_ && export ROX_CENTRAL_ADDRESS=_<address>:<port_number>_ <1>
----
+
<1> For `<rox_password>`, specify the {rh-rhacs-first} administrator password.

* Alternatively, you can configure the `ROX_API_TOKEN` and `ROX_CENTRAL_ADDRESS` environment variables.

** Run the following command to set the `ROX_API_TOKEN` and `ROX_CENTRAL_ADDRESS` environment variables:
+
[source,terminal,subs="+quotes"]
----
$ export ROX_API_TOKEN=_<api_token>_ && export ROX_CENTRAL_ADDRESS=_<address>:<port_number>_
----

> 

Reviewer: **thought:** To me these look like steps they need to take to satisfy the prerequisites, not actual prerequisites. Maybe we can move them in the Procedure section and remove this prerequisite item `You have configured the `ROX_PASSWORD` and `ROX_CENTRAL_ADDRESS` environment variables.`

Reviewer: I think e should list as a prerequisites, they have admin password or an API tokenm, and Central address.

.Procedure

* Run the following command to generate a diagnostic bundle:
+
[source,terminal]
----
$ roxctl -e "$ROX_CENTRAL_ADDRESS" -p "$ROX_PASSWORD" central debug download-diagnostics
----

* Alternatively, you can generate a diagnostic bundle using the following command:
+
[source,terminal]
----

> $ roxctl -e "$ROX_CENTRAL_ADDRESS" central debug download-diagnostics

Reviewer: **thought:** I think they can run this if they have already set an API token. So maybe take the steps from prerequisites and put them here.

----

//[retaining old content for reference]

> ////

Reviewer: Please remove the old content from the file, before merging.

// Module included in the following assemblies:
//
// * configuration/generate-diagnostic-bundle.adoc
:_content-type: PROCEDURE
[id="generate-diagnostic-bundle-using-roxctl-cli_{context}"]
= Generating a diagnostic bundle by using the roxctl CLI

You can generate a diagnostic bundle with the {rh-rhacs-first} administrator password or API token and central address by using the `roxctl` CLI.

.Prerequisites
* To generate a diagnostic bundle, you need `read` permission for the `Administration` resource. This is required for versions of the `DebugLogs` resource older than version 3.73.0.

* You must have configured the {product-title-short} administrator password or API token and central address.

.Procedure

> * To generate a diagnostic bundle by using the {product-title-short} administrator password

Reviewer 2: If there is a hard return after password, please remove

> * To generate a diagnostic bundle by using the {product-title-short} administrator password

Reviewer 2: ```suggestion
* To generate a diagnostic bundle by using the {product-title-short} administrator password,
```

perform the following steps:

. Run the following command to configure the `ROX_PASSWORD` and `ROX_CENTRAL_ADDRESS` environment variables:
+
[source,terminal,subs="+quotes"]
----
$ export ROX_PASSWORD=_<rox_password>_ && export ROX_CENTRAL_ADDRESS=_<address>:<port_number>_ <1>
----
+
<1> For `<rox_password>`, specify the {product-title-short} administrator password.

. Run the following command to generate a diagnostic bundle by using the {product-title-short} administrator password:
+
[source,terminal]
----
$ roxctl -e "$ROX_CENTRAL_ADDRESS" -p "$ROX_PASSWORD" central debug download-diagnostics
----

> * To generate a diagnostic bundle by using the API token perform the following steps:

Reviewer 2: ```suggestion
* To generate a diagnostic bundle by using the API token, perform the following steps:
```

// Module included in the following assemblies:
//
// * configuration/generate-diagnostic-bundle.adoc
:_content-type: PROCEDURE
[id="generate-diagnostic-bundle-using-roxctl-cli_{context}"]
= Generating a diagnostic bundle by using the roxctl CLI

You can generate a diagnostic bundle with the {rh-rhacs-first} administrator password or API token and central address by using the `roxctl` CLI.

[discrete]
== Prerequisites

> * If you are using {product-title-short} version 3.73 or older you need `read` permission for the `Administration` resource.

Author: @janisz @dhaus67 Is this sentence technically correct?

Author: Got it, thanks!

// Module included in the following assemblies:
//
// * configuration/generate-diagnostic-bundle.adoc
:_content-type: PROCEDURE
[id="generate-diagnostic-bundle-using-roxctl-cli_{context}"]
= Generating a diagnostic bundle by using the roxctl CLI

You can generate a diagnostic bundle by using the `roxctl` CLI.

.Prerequisites

> * To generate a diagnostic bundle, you need `read` permission for the `Administration` resource. This is required for versions of the `DebugLogs` resource older than version 3.73.0.

Reviewer 2: I think this is what is meant, but please verify with Eng.

* You have configured the `ROX_PASSWORD` and the `ROX_CENTRAL_ADDRESS` environment variables.

> ** Set the `ROX_PASSWORD` and the `ROX_CENTRAL_ADDRESS` environment variables:

Reviewer 2: ```suggestion
** Set the `ROX_PASSWORD` and the `ROX_CENTRAL_ADDRESS` environment variables by running the following command:
```

Reviewer 2: (suggested because of OpenShift style)

Author: Done

+
[source,terminal,subs="+quotes"]
----
$ export ROX_PASSWORD=_<rox_password>_ && export ROX_CENTRAL_ADDRESS=_<address>:<port_number>_ <1>
----
+

> <1> For `<rox_password>`, specify the stackrox administration password.

Reviewer 2: ```suggestion
<1> For `<rox_password>`, specify the {product-title-short} administrator password.
```

Reviewer 2: @rukletsov please verify that stackrox administration password = RHACS administrator password? (We seem to use simply "administrator password" in the current docs, but maybe adding RHACS makes it more clear.)

Author: Done

.Procedure
* Run the following command to generate a diagnostic bundle:
+
[source,terminal]
----
$ roxctl -e "$ROX_CENTRAL_ADDRESS" -p "$ROX_PASSWORD" central debug download-diagnostics

> ----

Author: @rukletsov @janisz Please can you review this revised content, thanks!

> ----

Author: @dhaus67 technical review of the prerequisites and procedure steps, please!

> ----

Author: @dhaus67 Please do a technical review of only this portion of the content, every other content you see below this comment is commented out (meaning it won't appear in the docs portal), thanks!

Author: Please suggest any changes over here.

Reviewer 3: So, if I understand correctly, only the `ROX_PASSWORD` way will currently be shown. I feel like we should show both (so, on one side using `ROX_API_TOKEN` environment variable, as well as the `ROX_PASSWORD / -p` way.

Otherwise, the contents look good to me.

Author: Done.

Author: @dhaus67 Please view the changes here: https://github.com/openshift/openshift-docs/pull/56709/commits/9f0985d2dbe28ffcf142411c77ba4bd0ffb70b17

////
.Prerequisites
* To generate a diagnostic bundle, you need `read` permission for the `DebugLogs` resource.

* You have configured the `ROX_API_TOKEN` and the `ROX_CENTRAL_ADDRESS` environment variables.

** Set the `ROX_API_TOKEN` and the `ROX_CENTRAL_ADDRESS` environment variables:
+
[source,terminal,subs="+quotes"]
----
$ export ROX_API_TOKEN=_<api_token>_
----
+
[source,terminal,subs="+quotes"]
----
$ export ROX_CENTRAL_ADDRESS=_<address>:<port_number>_
----
////

////
.Procedure
* Run the following command to generate a diagnostic bundle:
+
[source,terminal]
----
$ roxctl -e "$ROX_CENTRAL_ADDRESS" central debug download-diagnostics
----
+
[source,terminal]
----

> $ roxctl central debug download-diagnostics

Reviewer 3: Shouldn't this be dropped, now that we have defined the command in l.57 ($ roxctl -e "$ROX_CENTRAL_ADDRESS" central debug download-diagnostics)?

Right here, the `-e` flag is missing so this will be incomplete.

Author: > Shouldn't this be dropped, now that we have defined the command in l.57 ($ roxctl -e "$ROX_CENTRAL_ADDRESS" central debug download-diagnostics)?
> 
> Right here, the `-e` flag is missing so this will be incomplete.

@dhaus67 This command is obsolete, please see https://github.com/openshift/openshift-docs/pull/56709/files#:~:text=Resolve%20conversation-,.Procedure,%2D%2D%2D%2D,-Member

.Prerequisites

> * To generate a diagnostic bundle, you need `read` permission for the `DebugLogs` resource.

Reviewer 4: ```suggestion
* To generate a diagnostic bundle, you need `read` permission for the `Administration` resource, prior to version 3.73.0 `DebugLogs` resource.
```

Author: Done.

* You have configured the `ROX_API_TOKEN` and the `ROX_CENTRAL_ADDRESS` environment variables.

> ** Set the `ROX_API_TOKEN` and the `ROX_CENTRAL_ADDRESS` environment variables:

Reviewer 4: Instead of using `ROX_API_TOKEN`, this is also possible with `-p "$ROX_PASSWORD"`

Author: @rukletsov IIUC, all instances of `ROX_API_TOKEN` can be replaced with `ROX_PASSWORD` and the export command would be:
`$ export ROX_PASSWORD=_<rox_password>_` instead of `$ export ROX_API_TOKEN=_<api_token>_`

Is that correct?

Reviewer 5: example with password https://github.com/stackrox/stackrox/blob/e2c428cfdb8d7d3d542fb8e247962e0402a8cb32/scripts/ci/lib.sh#L159

It's worth to mention that password is admin password

Author: Done.
